internal/route/router.go

```diff
@@ -142,7 +168,7 @@ func (r *SimpleRouter) routeToConfig(
 		return nil, nil, err
 	}
 
-	httpReq, err := http.NewRequestWithContext(context.Background(), "POST", configToRoute.HTTPURL, bytes.NewReader(bodyBytes))
+	httpReq, err := http.NewRequestWithContext(ctx, "POST", configToRoute.HTTPURL, bytes.NewReader(bodyBytes))
```

Comment:
For my understanding - what do we get out of passing the context in here?

Reply:
Copying from godocs:
```
// Package context defines the Context type, which carries deadlines,
// cancellation signals, and other request-scoped values across API boundaries
// and between processes.
//
// Incoming requests to a server should create a Context, and outgoing
// calls to servers should accept a Context. The chain of function
// calls between them must propagate the Context, optionally replacing
// it with a derived Context created using WithCancel, WithDeadline,
// WithTimeout, or WithValue. When a Context is canceled, all
// Contexts derived from it are also canceled.
```

`context` allows you to pass the state of the request to different goroutines/functions during the life of that request. For example, it allows for things like when the connection terminates, other goroutines/processes to abort. For us we're currently only using `context` to pass around the value of `client`.

```
	// Use context values only for request-scoped data that transits
	// processes and API boundaries, not for passing optional parameters to
	// functions.
```

Reply:
It seems like `http.NewRequestWithContext` won't do anything with our `client` context, but it's good practice to pass it around?

Reply:
yeah that's my understanding